[docs]copilot (#6310)
ToolJet/ToolJet: docs/versioned_docs/version-2.5.0/copilot.md

---
id: tooljet-copilot
title: Copilot
---

:::info BETA
Tooljet Copilot is currently available in public beta for all the users.
:::

**ToolJet Copilot** helps you write your queries faster. It uses OpenAI to suggest queries based on your data. 

## Activating Copilot

> To access the Copilot, each user, including administrators, can sign up for the waitlist program individually. After signing up, each user will receive a unique API key associated with their account. **Only administrators**, however, will have the privilege to enable Copilot within the workspace settings. Administrators can navigate to the Copilot section in the workspace settings and toggle on the "Enable Copilot" option using their respective API key.

Author: ```suggestion
To gain access to Copilot, all users, including administrators, can individually sign up for the waitlist program. Upon successful sign up, each user will be issued a distinctive API key linked to their account. However, the exclusive authority to activate Copilot within the workspace settings lies solely with administrators. Administrators can proceed to the Copilot section in the workspace settings, where they can utilize their respective API key to set the "Enable Copilot" toggle option.
```

You can join the waitlist here: **https://tooljet.com/copilot**

:::info
- Copilot can be used by users who have **permissions to edit** the app with a copilot setup.
:::

<div style={{textAlign: 'center'}}>

<img className="screenshot-full" src="/img/copilot/enable.png" alt="ToolJet Copilot" />

</div>

## Using Copilot

Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.

> For Copilot to operate, it requires a plain English prompt. Copilot possesses the ability to interact with and access components, along with generating code for specific actions related to those components. This capability is made possible by Copilot's awareness of the application's current state.

Author: ```suggestion
Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.
```

### Example: Generating a SQL statement 

> We will demonstrate the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.

Author: ```suggestion
This example demonstrates the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
```

> - Let's create a RESTAPI query using this mock api endpoint: https://fakestoreapi.com/users/1

Author: ```suggestion
- Create a RESTAPI query using this mock api endpoint: https://fakestoreapi.com/users/1
```

---
id: tooljet-copilot
title: Copilot
---

:::info BETA
Tooljet Copilot is currently available in public beta for organizations.
:::

**ToolJet Copilot** helps you write your queries faster. It uses OpenAI to suggest queries based on your data. 

## Activating Copilot

> To activate Copilot for a ToolJet workspace, the administrator must go to the **workspace settings**, then the **Copilot** section, and toggle on the "**Enable Copilot**" option. Once Copilot is enabled, an **API key** must be provided in order to activate the feature.

Reviewer: ```suggestion
To activate Copilot for a ToolJet workspace, it is necessary to acquire an API key. Once the API key is obtained, Copilot can be enabled by the administrator via the workspace settings. Additionally, users with edit permissions for apps within the workspace can also utilize Copilot.
```

You can join the waitlist here: **https://tooljet.com/copilot**

:::info

> - A **new API key** will be needed for each workspace, with *only one user (admin)* allowed per workspace.

Reviewer: can remove this
